Game 1 FlappyBird/FlappyBird/Source/FlappyBird/Private/Character/FlappyCharacter.cpp

```diff
@@ -0,0 +1,53 @@
+﻿// 
+
+
+#include "FlappyBird/Public/Character/FlappyCharacter.h"
+
+#include "EnhancedInputComponent.h"
+#include "EnhancedInputSubsystems.h"
+#include "Camera/CameraActor.h"
+#include "Components/CapsuleComponent.h"
+#include "GameFramework/CharacterMovementComponent.h"
+
+AFlappyCharacter::AFlappyCharacter()
+{
+	PrimaryActorTick.bCanEverTick = true;
+
+	GetCapsuleComponent()->SetSimulatePhysics(true);
+	GetCapsuleComponent()->SetEnableGravity(true);
+}
+
+void AFlappyCharacter::BeginPlay()
+{
+	Super::BeginPlay();
+	check(FlappyContext);
+
+	APlayerController* PlayerController = Cast<APlayerController>(GetController());
+	check(PlayerController);
+	if(UEnhancedInputLocalPlayerSubsystem* Subsystem = ULocalPlayer::GetSubsystem<UEnhancedInputLocalPlayerSubsystem>(PlayerController->GetLocalPlayer()))
+	{
+		Subsystem->AddMappingContext(FlappyContext, 0);
+	}
+
+	PlayerController->SetViewTargetWithBlend(GameCamera);
+}
+
+void AFlappyCharacter::Flap(const FInputActionValue& Value)
+{
+	GetCapsuleComponent()->AddImpulse(FVector(0,0,FlapImpulse));
```

Comment:
There's a launch that doesn't need to get a component to work.

Reply:
Yeah I just decided to do an impulse